lib/spack/spack/stage.py

```diff
@@ -388,7 +387,7 @@ def save_filename(self):
             # This prefers using the URL associated with the default fetcher if
             # available, so that the fetched resource name matches the remote
             # name
-            return possible_filenames[0]
+            return sup.ensure_legal_path(possible_filenames[0])
```

Comment:
Can this move to just formatting `os.path.basename(self.default_fetcher.url)` in `expected_archive_files`? I think that is more clear about where the problem is occurring (while what you did here should technically always work, it doesn't always need to be applied to `possible_filenames[0]`).

You could also apply it to other paths in that function but e.g. we generate `.mirror_paths` ourselves, so it isn't the same problem as using the URL basename.

Reply:
Sure, the less we touch the smaller the potential for issues later, I'll make the change.